### PR TITLE
XCMS tools for plotting EIC and Peaks

### PR DESCRIPTION
**Pull request by wverastegui**
This PR contains the XCMS plotting tools for peaks and EIC.

- Updated the help text
- Updated the test for mzML
- Renamed outputname, it inlcudes <m/z>, <rt> and <dataset_id>
- Fixed plot cutoff of the image
- Updated test plots

closes https://github.com/RECETOX/galaxytools/issues/498 & https://github.com/RECETOX/galaxytools/issues/499

### CODE REVIEW

**Comment by wverastegui** [comment]
This [commit](https://github.com/RECETOX/galaxytools/pull/523/commits/a0d352dacabe38e6f7a362221ac8ad198a8e540b) fixed the issue of the left margin cutoff.

File: tools/xcms/xcms_plot_raw.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+<tool id="xcms_plot_raw" name = "Plot filtered by m/z and retention time" version="@TOOL_VERSION@+galaxy0" profile="21.09">
```
Comment (hechth) [inline tools/xcms/xcms_plot_raw.xml:1]:
```suggestion
<tool id="xcms_plot_raw" name = "xcms plot raw" version="@TOOL_VERSION@+galaxy0" profile="21.09">
```

File: tools/xcms/xcms_plot_raw.xml
Comment (hechth) [inline tools/xcms/xcms_plot_raw.xml:1]: please add the crossref to the bio.tools id the XCMS package to the tool

File: tools/xcms/xcms_plot_eic.xml
Comment (hechth) [inline tools/xcms/xcms_plot_eic.xml:1]: please add the bio.tools id of XCMS as crossref - see other tools like matchms as an example

File: tools/xcms/xcms_plot_eic.xml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+png(filename = '${output_filename}')
+plot(chr)
+dev.off()
+        </configfile>
+    </configfiles>
+    <inputs>
+        <param type="data" name="input" format="mzML" label="Input mzML file"/>
+        <param type="float" name="mz_value" value="0" label="m/z Value" help="m/z value for the EIC"/>
+        <param type="integer" value="10" name="tolerance_ppm" label="Tolerance (ppm)" help="Tolerance for m/z value in ppm"/>
+        <param type="integer" value="1" name="mslevel" label="MS Level" help="MS level for the EIC"/>
```
Comment (hechth) [inline tools/xcms/xcms_plot_eic.xml:30]: Please add reasonable minimum values to all these parameters

File: tools/xcms/xcms_plot_raw.xml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+plot(raw)
+dev.off()
+        </configfile>
+    </configfiles>
+    <inputs>
+        <param type="data" name="input" format="mzml" label="Input mzML file"/>
+        <param type="float" name="mz_value" value="0.0" label="m/z Value" help="m/z value for the plot"/>
+        <param type="integer" name="tolerance_ppm" value="10" label="Tolerance (ppm)" help="Tolerance for m/z value in ppm"/>
+        <param type="float" name="rt" label="Retention Time" value="0.0" help="Retention time for the plot"/>
+        <param type="float" name="rt_range" value="5.0" label="Retention Time Range" help="Retention time range for the plot"/>
```
Comment (hechth) [inline tools/xcms/xcms_plot_raw.xml:35]: please add reasonable minimum values to these parameters

**Review by hechth — CHANGES_REQUESTED** [review]
Please see the comments and re-request review from me and @KristinaGomoryova after adressing them.

**Comment by hechth** [comment]
@yguitton should we put those tools rather to the other w4m XCMS tools into your repo, what do you think?
@jorainer these tools add some of the plotting capabilities described in your XCMS tutorial to Galaxy. If you make it referenceable we can link it here so that people see the further functions provided by the package :)

File: tools/xcms/xcms_plot_eic.xml
Comment (wverastegui) [reply tools/xcms/xcms_plot_eic.xml:1]: Done -> [710ef17](https://github.com/RECETOX/galaxytools/pull/523/commits/710ef1787d3209d01cd76442a1c7d8db239cde7c)

File: tools/xcms/xcms_plot_raw.xml
Comment (wverastegui) [reply tools/xcms/xcms_plot_raw.xml:1]: Done -> [710ef17](https://github.com/RECETOX/galaxytools/pull/523/commits/710ef1787d3209d01cd76442a1c7d8db239cde7c)

File: tools/xcms/xcms_plot_raw.xml (hunk 3 above)
Comment (wverastegui) [reply tools/xcms/xcms_plot_raw.xml]: Added -> [3944629](https://github.com/RECETOX/galaxytools/pull/523/commits/39446294577066fc66e03ec19dce25f86c645aed)

File: tools/xcms/.shed.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+description: XCMS and Spectra plotting tools
+long_description: |
+    
+type: unrestricted
+auto_tool_repositories:
+  name_template: "{{ tool_id }}"
+  description_template: "{{ tool_name }} tool from the xcms package"
+suite:
+  name: suite_xcms_plot
+  description: tools from xcms are used for visualization of raw data
```
[the remark this answers is not in the document]
Comment (wverastegui) [reply tools/xcms/.shed.yml]: added -> [6f298df](https://github.com/RECETOX/galaxytools/pull/523/commits/6f298df23c2b4e50aa967187995ced23dc1f4545)

File: tools/xcms/.shed.yml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+name: xcms_plot
+owner: recetox
+remote_repository_url: "https://github.com/RECETOX/galaxytools/tree/master/tools/plot_eic"
+homepage_url: "https://github.com/RECETOX/galaxytools"
+categories:
+  - Metabolomics
+description: XCMS and Spectra plotting tools
+long_description: |
+
```
[the remark this answers is not in the document]
Comment (wverastegui) [reply tools/xcms/.shed.yml]: Added -> [6f298df](https://github.com/RECETOX/galaxytools/pull/523/commits/6f298df23c2b4e50aa967187995ced23dc1f4545)

File: tools/xcms/xcms_plot_raw.xml
Comment (hechth) [reply tools/xcms/xcms_plot_raw.xml:1]: These are not the minimum values but defaults.

**Comment by hechth** [comment]
@yguitton then I will close this here and open a new PR on w4m - should we make a new folder for these like `xcms_plot` or put them to the existing folder?

**Pull request closed without merging**